public/app/features/alerting/unified/components/rule-editor/CloudAlertPreview.tsx

```diff
@@ -0,0 +1,154 @@
+import { css } from '@emotion/css';
+import { max } from 'lodash';
+import React from 'react';
+
+import { DataFrame, GrafanaTheme2 } from '@grafana/data/src';
+import { Icon, TagList, Tooltip, useStyles2 } from '@grafana/ui/src';
+
+import { GrafanaAlertState, isGrafanaAlertState, Labels } from '../../../../../types/unified-alerting-dto';
+import { labelsToTags } from '../../utils/labels';
+import { AlertStateTag } from '../rules/AlertStateTag';
+
+interface AlertPreviewInstance {
+  state: GrafanaAlertState;
+  info?: string;
+  labels: Labels;
+}
+
+interface AlertPreview {
+  instances: AlertPreviewInstance[];
+}
+
+function mapDataFrameToAlertPreview({ fields }: DataFrame): AlertPreview {
```

Comment:
Can we add a code comment here why we need this function or why you would use it?